### PR TITLE
NMS-13187 Add Open API doc and Swagger UI to v2 RESTful API

### PR DESCRIPTION
**Pull request by jadyhou**
Use the URL below can get to the standard openapi doc page:
http://localhost:8980/opennms/api/v2/openapi.json

Use the URL below can get to the swagger ui page
http://localhost:8980/opennms/api/v2/api-docs/?url=/opennms/api/v2/openapi.json#/

All v2 restful endpoints are automatically scanned:

We can make rest call on the swagger UI directly:

### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13187

### CODE REVIEW

**Comment by RangerRick** [comment]
This looks great at first glance, I'll try playing around with the branch today.

File: opennms-webapp-rest/pom.xml [hunk 1]
```diff
@@ -399,6 +399,22 @@
       <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.cxf</groupId>
+      <artifactId>cxf-rt-rs-service-description-openapi-v3</artifactId>
+      <version>3.2.4</version>
```
Comment (cgorantla) [inline opennms-webapp-rest/pom.xml:405]: version can be defined in parent pom.xml and can be referenced here.

File: opennms-webapp-rest/pom.xml [hunk 2]
```diff
@@ -399,6 +399,22 @@
...
+        <exclusion>
+          <groupId>org.apache.cxf</groupId>
+          <artifactId>cxf-rt-frontend-jaxrs</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.webjars</groupId>
+      <artifactId>swagger-ui</artifactId>
+      <version>3.13.0</version>
```
Comment (cgorantla) [inline opennms-webapp-rest/pom.xml:416]: same as above.

**Review by cgorantla — COMMENTED** [review]
Looks pretty good to me.

File: opennms-webapp-rest/pom.xml (hunk 1 above)
Comment (RangerRick) [reply opennms-webapp-rest/pom.xml:405]: ah, good catch

Comment (jadyhou) [reply opennms-webapp-rest/pom.xml]: Good catch. Updated.

File: opennms-webapp-rest/pom.xml (hunk 2 above)
Comment (jadyhou) [reply opennms-webapp-rest/pom.xml]: Updated.

File: pom.xml [hunk 3]
```diff
@@ -1525,6 +1525,10 @@
     <ci.rerunFailingTestsCount>0</ci.rerunFailingTestsCount>
     <!-- Limit ITs to 30 minutes by default -->
     <test.fork.timeout>1800</test.fork.timeout>
+
+    <!-- OpenAPI-->
+    <cxfServiceDescriptionOpenapiVersion>3.2.4</cxfServiceDescriptionOpenapiVersion>
```
Comment (cgorantla) [inline pom.xml:1530]: nit:  Place it in alphabetical order in above list. don't need to separate it out.

Comment (jadyhou) [reply pom.xml]: Updated.

**Review by RangerRick — CHANGES_REQUESTED** [review]
Looks great!

One thing it would be nice to do is change the default `.json` file used when you hit the `/api-docs/` URL instead of serving up the petstore example.  Kind of a pain that it appears to be _in_ the swagger-ui bundle, but I did find [this](https://stackoverflow.com/a/55047755) which appears to be a hacky way to work around it.

**Review by RangerRick — APPROVED** [review]

**Comment by RangerRick** [comment]
> One thing it would be nice to do is change the default `.json` file used when you hit the `/api-docs/` URL instead of serving up the petstore example. Kind of a pain that it appears to be _in_ the swagger-ui bundle, but I did find [this](https://stackoverflow.com/a/55047755) which appears to be a hacky way to work around it.

Just for posterity, I'll post this here, there's more to be done to make this work right, so not gonna let this hold up the PR. We can look into properly making it go to the right default place in a separate issue for doing branding and such.

**Comment by jadyhou** [comment]
> > One thing it would be nice to do is change the default `.json` file used when you hit the `/api-docs/` URL instead of serving up the petstore example. Kind of a pain that it appears to be _in_ the swagger-ui bundle, but I did find [this](https://stackoverflow.com/a/55047755) which appears to be a hacky way to work around it.
> 
> Just for posterity, I'll post this here, there's more to be done to make this work right, so not gonna let this hold up the PR. We can look into properly making it go to the right default place in a separate issue for doing branding and such.

Yeah, tried on Friday and didn't make it work to change the hard-coded line in swagger UI. I'll write up an enhancement Jira for this : ).

**Pull request merged**